### ISSUE TITLE
Using /// breaks markdown headings

### ISSUE THREAD

**Issue by grantroch** [lawmurray/doxide#41]
Using markdown headers within a doc-comment does not render the header correctly. This is very similar to https://github.com/lawmurray/doxide/issues/38 in that using `/** ... */` works but `///` does not but for the header instead. Below the same example code as in that case.

```
namespace example {

/** First function
 *
 * This one uses block comments and multiple paragraphs and such can be used.
 *
 * #### Example
 *
 * ```c++
 * first();
 * ```
 */
void first();

/// Second function
///
/// This one uses single-line comments and the paragraphs are broken.
///
/// #### Example
///
/// ```c++
/// second();
/// ```
void second();

} // namespace example
```
This renders the header correctly for `first` but not for `second`. From what I can see, the only difference in the rendered markdown is an extra space before the markdown in the instance that doesn't work.
![image](https://github.com/lawmurray/doxide/assets/59003781/790f7ddc-4a5e-4ff8-8a2d-3519b4a8fae8)

**Comment by lawmurray** [comment]
Thanks for the report @grantroch. I've reproduced this and am looking into it.

**Comment by lawmurray** [comment]
A little more testing to come, but generally have this fixed now along with other related issues (e.g. admonitions had similar problems).

**Comment by lawmurray** [comment]
Thanks again for this report @grantroch . Fixes our now incorporated into v0.4.5 and released.

**Comment by grantroch** [comment]
Thank you, @lawmurray!
